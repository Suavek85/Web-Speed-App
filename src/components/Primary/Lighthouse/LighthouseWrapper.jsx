import React, { useState } from 'react'
import { useSelector } from "react-redux"
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'

import LighthouseTabContent from './LighthouseTabContent/LighthouseTabContent'
import LighthouseTabs from './LighthouseTabs/LighthouseTabs'
import LighthouseHeader from './LighhouseHeader/LighthouseHeader'
import { tabsDescriptions, tabsTitles } from '../../../constants/content'
import styles from './Lighthouse.scss'
import { tabletWidth, smallMobileWidth } from'../../../scss/mediaqueries'

export default function LightHouse() {
  
  const [ activeTab, setActiveTab ] = useState('0')
  const activeTabInt = parseInt(activeTab)

  const handleActiveTab = e => setActiveTab(e.target.id)

  const activeTabDesc = tabsDescriptions[activeTabInt]
  const activeTabTitle = tabsTitles[activeTabInt]

  const getDataState = useSelector(state => state.stateReducer.getLighthouseData)
  const { score } = getDataState[activeTabInt]

  //STYLES
  const cx = classNames.bind(styles)
  const getWrapperClass = cx({
    block: true,
    blockTablet: useMediaQuery(tabletWidth),
    blockSmallMobile: useMediaQuery(smallMobileWidth),
  })

  //PROPS
  const tabsProps = { handleActiveTab, activeTab }

  const tabContentProps = { 
    activeTabDesc,
    activeTabTitle,
    score,
  }

  return (
    <section className={ getWrapperClass } >
      <LighthouseHeader />
      <div className={ styles.blockContent }>
        <LighthouseTabs {...tabsProps} />
        <LighthouseTabContent  {...tabContentProps} />
      </div>
    </section>
  )
}