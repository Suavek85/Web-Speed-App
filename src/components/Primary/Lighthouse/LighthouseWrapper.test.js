import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import LighthouseWrapper from './LighthouseWrapper'

describe('LighthouseWrapper snapshot', () => {

  it('should match the snapshot', () => {
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    spyUseSelector.mockReturnValue(null)
    const wrapper = shallow(<LighthouseWrapper />)
    expect(wrapper).toMatchSnapshot()
  })

})