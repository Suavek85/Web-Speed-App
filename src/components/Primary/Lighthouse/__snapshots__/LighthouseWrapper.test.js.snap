// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LighthouseWrapper snapshot should match the snapshot 1`] = `
<section
  className="block"
>
  <LighthouseHeader />
  <div>
    <LighthouseTabs
      activeTab="0"
      handleActiveTab={[Function]}
    />
    <LighthouseTabContent
      activeTabDesc="First Contentful Paint marks the time at which the first text or image is painted."
      activeTabTitle="First Contentful Paint"
      score={null}
    />
  </div>
</section>
`;
